### PR TITLE
CAS-557: Modify list proposals endpoint to allow filtering by status

### PR DESCRIPTION
**Pull request by 0xmovses**
•updates `Get CommunityProposalsForUser` to accept filter param, adjusts SQL accordingly 
•adds two tests `Get user proposals` and `Get user proposals with 'draft' status filter applied` for basic cases. 

N.B Merging this into `rich/cas-555` as they contain dependant code.

### CODE REVIEW

File: backend/main/server/helpers.go [hunk 1]
```diff
@@ -517,6 +517,7 @@ func (h *Helpers) createProposal(p models.Proposal) (models.Proposal, errorRespo
 		return models.Proposal{}, errStrategyNotFound
 	}
 
+	fmt.Printf("Proposal strategy: %v \n", p)
```
Comment (jbluks) [inline backend/main/server/helpers.go:520]: Remove

**Review by jbluks — COMMENTED** [review]
Are these changes right? It looks like these changes are all related to creating proposals, not modifying the list end point. Maybe just needs to be updated with release-1.6.0?

**Comment by rvmelkonian** [comment]
ready for review now

File: backend/main/server/controllers.go [hunk 2]
```diff
@@ -350,10 +351,17 @@ func (a *App) getDraftProposal(w http.ResponseWriter, r *http.Request) {
 func (a *App) getUserProposals(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
 	addr := vars["addr"]
+	filter := r.FormValue("filter")
 
 	pageParams := getPageParams(*r, 25)
 
-	communities, totalRecords, err := models.GetCommunityProposalsForUser(a.DB, addr, pageParams)
+	fmt.Println("filter", filter)
```
Comment (jbluks) [inline backend/main/server/controllers.go:358]: Remove

**Review by jbluks — APPROVED** [review]
Looks good - remove one println

**Pull request merged**